libs/blocks/email-collection/email-collection.js

```diff
@@ -0,0 +1,513 @@
+    await submitForm(form);
+    setMaxHeightToForm(text, true);
+  });
+}
+
+function transformCtaToBtn(el) {
+  const buttonTypes = ['submit', 'close-form'];
+  const link = el.querySelector('a');
+  const type = new URL(link.href).hash.replace('#', '');
+  if (!buttonTypes.includes(type)) return el;
+  const button = createTag(
+    'button',
+    {
+      class: link.className,
+      'aria-label': link.textContent,
+      ...(type === 'submit' && { type: 'submit', form: FORM_ID }),
+    },
+    link.innerHTML,
+  );
+
+  if (type === 'close-form') {
+    button.addEventListener('click', function closeForm() {
+      closeModal(this.closest('.dialog-modal'));
+    });
+  }
+
+  const buttonContainer = createTag('div', { class: 'button-container' });
+  buttonContainer.appendChild(button);
+  el.replaceWith(buttonContainer);
+
+  return buttonContainer;
+}
+
+function decorateSubscribedMessage(text) {
+  const showForm = text.querySelector('a[href$="#show-form"]');
+  if (!showForm) return;
+  const span = createTag(
+    'span',
+    {
+      class: 'show-form',
+      role: 'button',
+      tabindex: '0',
+    },
+    showForm.textContent,
+  );
+  const email = createTag('p', { class: 'subscribed-email hidden body-m' });
+
+  ['click', 'keydown'].forEach((eventType) => {
+    span.addEventListener(eventType, (e) => {
+      const { key, code } = e;
+      if (eventType === 'keydown' && key !== 'Enter' && code !== 'Space') return;
+      e.preventDefault();
+      showHideMessage({ hideMessage: true });
+    });
+  });
+
+  const parent = showForm.parentElement ?? span;
+  showForm.replaceWith(span);
+  parent.classList.replace('body-m', 'body-xs');
+  parent.before(email);
+}
+
+function decorateText(elChildren) {
+  const foreground = elChildren[0];
+  elChildren.forEach((child, childIndex) => {
+    const isForeground = child === foreground;
+    const text = isForeground ? child.lastElementChild : child.firstElementChild;
+    text.classList.add('text');
+    decorateButtons(child);
+    [...text.children].forEach((textEl) => {
+      if (textEl.classList.contains('action-area')) {
+        transformCtaToBtn(textEl);
+      } else if (textEl.childElementCount === 1 && textEl.firstElementChild.tagName === 'PICTURE') {
+        textEl.classList.add('icon-area');
+      } else if (['H1', 'H2', 'H3', 'H4', 'H5', 'H6'].includes(textEl.tagName)) {
+        textEl.classList.add('heading-l');
+      } else {
+        textEl.classList.add('body-m');
+      }
+    });
+
+    if (!isForeground) {
+      text.classList.add('hidden');
+      text.parentElement.remove();
+      foreground.appendChild(text);
+      if (childIndex === 1) decorateSubscribedMessage(text);
+    }
+  });
+}
+
+async function checkIsSubscribed() {
+  const { mpsSname } = getFormData('metadata');
+  const token = await getIMSAccessToken();
+  const { email } = await getIMSProfile();
+
+  try {
+    const submitFormReq = await fetch(getApiEndpoint('is-subscribed'), {
```

Comment:
You could create a custom fetch method that encompasses endpoint and body. I see:
* https://github.com/adobecom/milo/pull/4836/files#diff-f0bd62d49472be1efc907269ab70c884b6fc595fdc747898820a76ebc6c2ac5dR449
* https://github.com/adobecom/milo/pull/4836/files#diff-f0bd62d49472be1efc907269ab70c884b6fc595fdc747898820a76ebc6c2ac5dR200

they're pretty much the same, so you can use endpoint and body as arugments